adapters/from_sql.py

```diff
@@ -0,0 +1,194 @@
+    if item.end_date == None:
+      task.last_modified = to_timestamp(item.execution_date)
+    else:
+      task.last_modified = to_timestamp(item.end_date)
+    task.log_url = 'https://youtu.be/dQw4w9WgXcQ'  # TODO(dommarques): figure out how to get the log in here
+    if item.state == None:
+      item.state == 'none'
+    else:
+      task.error = item.state
+    task_objects.append(task)
+  return task_objects
+
+
+def get_task_info(dag_id, execution_date, airflow_db):
+  """Gets task-related info to fill in missing release object info.
+
+  Args:
+    execution_date: a python datetime.datetime
+    airflow_db: the database connection object
+
+  Returns:
+    most_recent_task: the most recent active task
+    state: the state of the release based on the task states
+  """
+  task_query = to_sql_tasks(dag_id, execution_date)
+  task_data = airflow_db.query(task_query)
+  task_objects = read_tasks(task_data)
+  state = 0
+  task_ids = []
+  if task_objects:
+    most_recent_task = task_objects[len(task_objects) - 1]
+  else:
+    most_recent_task = None
+  for task in task_objects:
+    if task.status > state:
+      state = task.status
+    task_ids.append(task.task_name)
+  return task_ids, most_recent_task, state
+
+
+def read_xcom_vars(xcom_data):
+  """Reads the xcom data to get the dict of vars and the green build SHA."""
+  # occasionally, the xcom data doesn't exist yet. This is a workaround
+  xcom_named_tuple = namedtuple('Row', ['xcom_dict', 'green_sha'])
+  try:
+    xcom_data = xcom_named_tuple._make(xcom_data)
+    # The index 0 is here because the data returns in a tuple
+    xcom_dict = json.loads(xcom_data.xcom_dict[0])
+    green_sha = xcom_data.green_sha[0]
+    return xcom_dict, green_sha
+  except IndexError:
+    return None, None
+  except TypeError:
+    return None, None
+
+
+def get_xcom(execution_date, dag_id, airflow_db):
+  """Gets relevant xcom data from SQL.
+
+    Args:
+      execution_date: int or float
+      dag_id: str
+      airflow_db: airflow database connection object
+
+    Returns:
+      xcom_dict: the dictionary of xcom information
+      green_sha: the green build SHA
+  """
+  xcom_query = to_sql_xcom(dag_id, execution_date)
+  xcom_data = airflow_db.query(xcom_query)
+  xcom_dict, green_sha = read_xcom_vars(xcom_data)
+  return xcom_dict, green_sha
+
+
+def construct_links(green_sha):
+  """Takes the green build sha and derives the repo links from that."""
+  # this does nothing right now, but in the future it will get the links for
+  # each release
+  return ['https://youtu.be/dQw4w9WgXcQ']
+
+
+def parse_dag_id(dag_id):
+  """Parses the dag_id for the release branch and type.
+
+  Args:
+    dag_id: string
+
+  Returns:
+    branch: string
+    release_type: string
+  """
+  underscore_count = 0
+  last_underscore = False
+  for i, char in enumerate(dag_id):
+    if char == '_':
+      underscore_count += 1
+    if underscore_count == 1 and not last_underscore:
```

Comment:
it seems this statement does not do anything. Unless you have an indentation issue at line 186. Note that this method does not return anything if underscore_count != 2

Reply:
Have you looked at str.find and str.partition, this might help you

Reply:
I think you can just do 
```python
info = dag_id.split('_')
if len(info) == 3:
  return info[1], info[2]
elif len(info) == 2:
  return info[0], info[2]
// handle not valid type
```